File: #0896.monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        
        int size = nums.size();
        bool isIncreasing = true;
        bool isDecreasing = true;
        for (int i = 0; i < size - 1; i++) {
            if (nums[i + 1] - nums[i] < 0) {
                isIncreasing = false;
            }

            if (nums[i + 1] - nums[i] > 0) {
                isDecreasing = false;
            }
            
            if (!isIncreasing && !isDecreasing) {
                return false;
            }
        }
        return isIncreasing || isDecreasing;
    }
};

File: #0896.monotonic-array.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isMonotonic = function(nums) {
    let isUp = false;
    let isDown = false;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i + 1] > nums[i]) {
            isUp = true;
        }

        if (nums[i + 1] < nums[i]) {
            isDown = true;
        }

        if (isUp && isDown) {
            break;
        }
    }

    return !(isUp && isDown);
};
